parfive/downloader.py

```diff
@@ -372,28 +374,120 @@ def callback(token, future, main_pb):
                                           total=get_http_size(resp))
                     else:
                         file_pb = None
-                    with open(str(filepath), 'wb') as fd:
-                        while True:
-                            chunk = await resp.content.read(chunksize)
-                            if not chunk:
-                                # Close the file progressbar
-                                if file_pb is not None:
-                                    file_pb.close()
-
-                                return str(filepath)
 
-                            # Write this chunk to the output file.
-                            fd.write(chunk)
+                    # setup asyncio queue and writer
+                    queue = asyncio.Queue()
```

Comment:
Can we give this queue a better name, like `write_chunk_queue` or something, it just took me till reading all the way down to the download worker to remember exactly what we are doing with this queue.